src/libs/antares/inifile/inifile.cpp

```
@@ -253,51 +168,119 @@ uint IniFile::Section::size() const
+    {
+        return true;
+    }
+    return false;
+} // namespace Antares
 
-    // Load the file
-    IO::File::Stream file;
-    if (file.open(filename))
+std::string getSectionName(std::string line)
+{
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("[]"));
+    return splitLine[1];
+}
+
+bool isProperty(std::string line)
+{
+    if (std::count(line.begin(), line.end(), '=') == 1)
     {
-        enum
-        {
-            lineHardLimit = 2048
-        };
+        return true;
+    }
+    return false;
+}
+
+std::pair<std::string, std::string> getKeyValuePair(std::string line)
+{
+    boost::trim(line);
 
-        auto* line = new char[lineHardLimit];
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("="));
+
+    return std::make_pair(splitLine[0], splitLine[1]);
+}
+
+bool isComment(std::string line)
+{
+    boost::trim_left(line);
+    if (boost::starts_with(line, "#") || boost::starts_with(line, ";"))
+    {
+        return true;
+    }
+    return false;
+}
 
-        IniFile::Section* lastSection = nullptr;
-        uint64_t read = 0;
+bool isEmpty(std::string line)
+{
+    boost::trim_left(line);
+    return line.empty();
+}
 
-        // analyzing each line
-        while (file.readline(line, lineHardLimit))
+bool IniFile::readStream(std::istream& in_stream)
+{
+    std::string line;
+    IniFile::Section* currentSection(nullptr);
+    uint64_t read = 0;
+    while (std::getline(in_stream, line))
+    {
+        read += line.size();
+        if (startingSection(line))
         {
-            lastSection = AnalyzeIniLine(pFilename, this, lastSection, line, read);
+            currentSection = addSection(getSectionName(line));
+            continue;
         }
 
-        delete[] line;
+        if (isProperty(line))
+        {
+            std::pair<std::string, std::string> pair = getKeyValuePair(line);
+            currentSection->add(pair.first, pair.second);
```

`currentSection->add(pair.first, pair.second);` : what if we're reading a (key, value) pair and we're not in a section (meaning : **currentSection** is still **nullptr**)  ?
We should raise an error in this case.

---

Done